utils/markdown.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package utils
+
+import (
+	"strings"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/extension"
+	astExt "github.com/yuin/goldmark/extension/ast"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/util"
+)
+
+// StripMarkdown remove some markdown syntax
+func StripMarkdown(markdown string) (string, error) {
+	md := goldmark.New(
+		goldmark.WithExtensions(extension.Strikethrough),
+		goldmark.WithRenderer(
+			renderer.NewRenderer(renderer.WithNodeRenderers(
+				util.Prioritized(newNotificationRenderer(), 500),
+			)),
+		),
+	)
+
+	var buf strings.Builder
+	if err := md.Convert([]byte(markdown), &buf); err != nil {
+		return markdown, err
```

Comment:
This is not correct. We should return `"", err` here. Only at the call-site, should the decision of using the original string in case of an error should be taken.